app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   before_action :set_user, only: %i[index create show]
   def index
-    @posts = @user.posts
+    @posts = @user.posts.paginate(page: params[:page], per_page: 3)
```

Comment:
- [ ] As it is a requirement for this project, `solve N+1 problem when fetching all posts and their comments for a user`, 
Something like so: `@posts = @user.posts.includes(:comments)`, please kindly fix that.

Reply:
@Mihndim2020 I am using bullet to track N + 1 problems, and this query doesn't have N+1 problem.